lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -17,11 +17,13 @@ def _line2d_seg_dist(p1, p2, p0):
     and intersection point lies within segment if u is between 0 and 1
     """
```

Comment:
Technically correct. However, the issue is more a logical one rather than a technical division-by-zero problem. Therefore, I'd go with the following and leave all the original code unchanged in place.

```suggestion

    If p1 and p2 are identical, the distance between them and p0 is returned.
    """
    if np.all(p1 == p2):
        return np.hypot(p1[0] - p0[0],  p1[1] - p0[1])
```

Reply:
Added the docs and removed the comment. (Kept the solution with intermediate results.)

Reply:
> Kept the solution with intermediate results.

@oscargus: What is the motivation for keeping this? It would be nice to give a reason if you dismiss a suggestion. I find the separation of the special case and using explicit values there more clear, but can be convinced otherwise.

Reply:
Just wanted to avoid duplicate code (and was a bit unsure what the effect of np.asarray was).

But I see your point about separation as well. And will try to motivate better in the future.